### PR TITLE
[IA-3033] Misc UX feedback on Analysis Tab

### PR DESCRIPTION
**Pull request by jdcanas**
https://broadworkbench.atlassian.net/browse/IA-3033

This encompassing the laundry list of changes mentioned in IA-3033 except the following two. This is currently blocking some usability testing, so it would be nice to get it out soon. 

```
- yellow for embedded delete modal
- Include additional metadata conditionally?
    Size, full path, author, created date
    R / RMD extension?
```

### CODE REVIEW

File: src/components/ComputeModal.js [hunk 1]
```diff
@@ -1348,6 +1348,7 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
             existingRuntime.tool === 'RStudio' ? h(SaveFilesHelpRStudio) : h(SaveFilesHelp)
           ])],
           [willRequireDowntime(), () => h(Fragment, [
+            existingRuntime.tool !== tool ? p([`By continuing, you will be changing the application of your cloud environment from ${existingRuntime.tool} to ${tool}.`]) : undefined,
```
Comment (kyuksel) [inline src/components/ComputeModal.js]: Should we bold-face the tool names so they are easier to catch on the message below?

image

**Comment by kyuksel** [comment]
While testing the PR, I've noticed an unexpected behaviour that doesn't seem to happen in `dev`. See the recording below for the steps to reproduce. 

The behaviour in `dev` is such that upon clicking to create a `Jupyter` runtime in the first workspace, `RStudio` is disabled in the second workspace. I thought this would be allowed and there may be a bug in `dev` too but the `dev` behaviour feels more acceptable to me than the `PR` behaviour. What do you think?

Note that the two distinct workspaces belong to the same billing project (`callisto-dev`). Haven't checked yet if this has anything to do with the bug.

https://user-images.githubusercontent.com/5438223/152607041-512967c1-4f68-490c-92e1-13745b048c9b.mov

**Comment by kyuksel** [comment]
Verified the following so far:

- Hide the top right widget and dot dot dot
- Hide notebooks tab when analysis tab is enabled
- Wording around switching from Jupyter to RStudio cloud env

**Comment by kyuksel** [comment]
Are the excluded items below tracked on another ticket (assuming they're still desired)?

```
- yellow for embedded delete modal
- Include additional metadata conditionally?
    Size, full path, author, created date
    R / RMD extension?
```

**Comment by kyuksel** [comment]
The PR description doesn't include

> cloud env wording in context bar modal should be changed to ‘Environment settings’, and cloud bolts icon to gear icon, with gray outline (matching context bar)

in the list of excluded items and I don't see the change implemented?

**Comment by kyuksel** [comment]
> top context bar should be in analysisLauncher/appLauncher

Not sure what this means 🤔

**Comment by kyuksel** [comment]
https://broadworkbench.atlassian.net/browse/IA-3033?focusedCommentId=56038

**Comment by jdcanas** [comment]
> The PR description doesn't include
> 
> > cloud env wording in context bar modal should be changed to ‘Environment settings’, and cloud bolts icon to gear icon, with gray outline (matching context bar)
> 
> in the list of excluded items and I don't see the change implemented?

That refers to this
image

**Comment by jdcanas** [comment]
> > top context bar should be in analysisLauncher/appLauncher
> 
> Not sure what this means 🤔

This is super messy I don't why some github comments allow responses directly :(.

This refers to this change:

(today)
image

(Changed):
image

**Comment by jdcanas** [comment]
> While testing the PR, I've noticed an unexpected behaviour that doesn't seem to happen in `dev`. See the recording below for the steps to reproduce.
> 
> The behaviour in `dev` is such that upon clicking to create a `Jupyter` runtime in the first workspace, `RStudio` is disabled in the second workspace. I thought this would be allowed and there may be a bug in `dev` too but the `dev` behaviour feels more acceptable to me than the `PR` behaviour. What do you think?
> 
> Note that the two distinct workspaces belong to the same billing project (`callisto-dev`). Haven't checked yet if this has anything to do with the bug.
> 
>  screen_recording_bug_220204.mov

I'm not able to reproduce this myself locally with both PPW and non-PPW workspaces locally or in dev... regardless this doesn't seem related to any changes in this ticket, all of which are small style-related changes (no API calls or disable logic is touched). Maybe make a bug ticket to track this if you're able to reproduce it again?

File: src/components/ComputeModal.js [hunk 2]
```diff
@@ -1348,6 +1348,7 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
             existingRuntime.tool === 'RStudio' ? h(SaveFilesHelpRStudio) : h(SaveFilesHelp)
           ])],
           [willRequireDowntime(), () => h(Fragment, [
+            existingRuntime.tool !== tool ? p(['By continuing, you will be changing the application of your cloud environment from ', span({ style: { fontWeight: 'bold' } }, existingRuntime.tool), ' to ', span({ style: { fontWeight: 'bold' } }, tool), '.']) : undefined,
```
Comment (kyuksel) [inline src/components/ComputeModal.js]: For emphasizing text, we use `strong`, which would be detected by a screen reader and also would eliminate custom styling:

```suggestion
            existingRuntime.tool !== tool ? p(['By continuing, you will be changing the application of your cloud environment from ', strong([existingRuntime.tool]), ' to ', span({ style: { fontWeight: 'bold' } }, tool), '.']) : undefined,
```

File: src/pages/workspaces/workspace/Analyses.js [hunk 3]
```diff
@@ -409,10 +409,14 @@ const Analyses = _.flow(
...
           onDismiss: () => {
             refreshAnalyses()
+            refreshRuntimes()
+            refreshApps()
             setCreating(false)
           },
           onSuccess: () => {
             refreshAnalyses()
+            refreshRuntimes()
+            refreshApps()
```
Comment (kyuksel) [inline src/pages/workspaces/workspace/Analyses.js:419]: Why do we need to `refreshRuntimes/Apps` in this case?

File: src/components/RuntimeManager.js [hunk 4]
```diff
@@ -347,7 +348,7 @@ export default class RuntimeManager extends PureComponent {
 
     const galaxyApp = getCurrentApp(tools.galaxy.appType)(apps)
 
-    return h(Fragment, [
+    return isAnalysisTabVisible() ? h(Fragment) : h(Fragment, [
```
Comment (kyuksel) [inline src/components/RuntimeManager.js]: `null` would make the intention more explicit:

```suggestion
    return isAnalysisTabVisible() ? null : h(Fragment, [
```

**Comment by kyuksel** [comment]
Still planning on replacing Cloud Env Modal's cloud bolt icon with a gear icon?

File: src/pages/workspaces/workspace/Analyses.js (hunk 3 above)
Comment (jdcanas) [reply src/pages/workspaces/workspace/Analyses.js:419]: The analysis modal closing should trigger a refresh, as the ondismiss/success is passed through to the ComputeModal/GalaxyModal/CromwellModal that can open and create a runtime/app. Previous behavior was creating an app from the `Create`  button in the analysis tab did not refresh the context bar

**Review by kyuksel — APPROVED** [review]
I love the UX improvements!

**Review by Qi77Qi — APPROVED** [review]

**Pull request merged**